### PR TITLE
Added tests for web errors when trying to retrieve image

### PR DESCRIPTION
**Pull request by pandu-k**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
adds a test

* **What is the current behavior?** (You can also link to an open issue here)
no tests for when getting an image results in a 404

* **What is the new behavior (if this is a feature change)?**
a test for when getting an image results in a 4xx or 5xx

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No 

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
Manual testing - checked against a non existent image 

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Comment by pandu-k** [comment]
unit tests: https://github.com/marqo-ai/marqo/actions/runs/4463973937

**Comment by pandu-k** [comment]
Rerunning unit tests: https://github.com/marqo-ai/marqo/actions/runs/4558166909

**Comment by pandu-k** [comment]
unit tests: https://github.com/marqo-ai/marqo/actions/runs/4625139527

File: tests/s2_inference/test_clip_utils.py [hunk 1]
```diff
@@ -41,3 +46,28 @@ def run():
...
+            bad_response.status_code = status_code
+            mock_get.return_value = bad_response
+
+            @mock.patch('requests.get', mock_get)
+            def run():
+                try:
+                    clip_utils.load_image_from_path(good_url, {})
+                    raise AssertionError
+                except PIL.UnidentifiedImageError:
+                    pass
```
Comment (wanliAlex) [inline tests/s2_inference/test_clip_utils.py]: do you think we should return the error code from the HTTP request and add it to the error?

**Review by wanliAlex — CHANGES_REQUESTED** [review]

File: tests/s2_inference/test_clip_utils.py (hunk 1 above)
Comment (pandu-k) [reply tests/s2_inference/test_clip_utils.py]: The error code is propagated here: https://github.com/marqo-ai/marqo/blob/mainline/src/marqo/s2_inference/clip_utils.py#L110

I'll add this to the test

Comment (wanliAlex) [reply tests/s2_inference/test_clip_utils.py]: Thanks! I was actually looking at the wrong function and didn't notice this error code propagation.

Comment (pandu-k) [reply tests/s2_inference/test_clip_utils.py]: added HTTP error code assertion in other test: https://github.com/marqo-ai/marqo/pull/402/files#diff-2bb102914165cddf1cfa00d25eb21234795d98c9b3c2f513ce4c14acae701a8eR70

**Comment by pandu-k** [comment]
unit tests: https://github.com/marqo-ai/marqo/actions/runs/4684662639

**Review by wanliAlex — APPROVED** [review]

**Pull request merged**